src/js/common/schemaform/FormIntroButtons.jsx

```diff
@@ -4,40 +4,17 @@ import PropTypes from 'prop-types';
 import ProgressButton from '../../common/components/form-elements/ProgressButton';
 
 class FormIntroButtons extends React.Component {
-  componentWillMount() {
-    // TODO: When the api is available to get a list of all the saved forms, query
-    //  for that instead and save it in the redux store
-
-    // Load the form data into the redux store if possible
-    // NOTE: This does not fill the data into the form yet; just grabs it and
-    //  saves it for later (when we want to fill in the blanks).
-    // TODO: Probably should wrap this in
-    // if (this.props.loggedIn) { ... }
-    this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
-  }
-
-  // Load the form data again if we weren't logged in before but are now.
-  componentWillReceiveProps(newProps) {
-    // Ugh. So verbose.
-    let wasLoggedIn;
-    let isLoggedIn;
-    try {
-      wasLoggedIn = this.props.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        // Probably couldn't read a property somewhere along the way
-        wasLoggedIn = false;
-      }
-    }
-    try {
-      isLoggedIn = newProps.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        isLoggedIn = false;
-      }
-    }
-    if (!wasLoggedIn && isLoggedIn) {
-      this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
+  componentWillReceiveProps = (newProps) => {
```

Comment:
`mapStateToProps` is a selector, and shouldn't have side effects either. `returnUrl` is set in the action creator, so you could do the routing in there. That might actually be clearer, though you do have to pass the router to the action creator, which feels strange to me (though I have done that before in apps).